src/Provider.js

```diff
@@ -1,40 +1,35 @@
-import { Children, Component, createContext, createElement } from "react"
+import React from "react"
 import { shallowEqual } from "./utils/utils"
 
-export const MobXProviderContext = createContext({})
+export const MobXProviderContext = React.createContext({})
 
-export class Provider extends Component {
-    static contextType = MobXProviderContext
-    static displayName = "MobXProvider"
-
-    constructor(props, context) {
-        super(props, context)
-        this.state = {
-            ...context,
+export function Provider(props) {
```

Comment:
I am also thinking that `React.memo` could be applied here. Considering props passed here will be some stores with stable reference integrity, it's just fine to prevent that re-render right away. And if someone has primitive value for a _store_ than Provider will re-render just fine (and then fail with that hard exception).

Reply:
@vkrol What about this? Not sure why it's marked as resolved...

Reply:
I was thinking that this comment is related to to the comment about getting rid of the stores set changing restriction.

Reply:
I do not think that we should make this change in this PR because it will be definitely a breaking change. Probably we can do it in the future.

Reply:
Yea, it's an actually bad idea ... https://github.com/mobxjs/mobx-react/issues/745#issuecomment-517925519 so nevermind.

Reply:
> No, because Provider depends on children which are always different (unless they're memoized by parent, which is very unlikely). In other words Provider never recieves the same props.

I forgot about this.

Reply:
> No, because Provider depends on children which are always different (unless they're memoized by parent, which is very unlikely). In other words Provider never recieves the same props.

I forgot about this.

Reply:
Me too :)